### DISCUSSION TITLE
Is `target_compatible_with` not supported?

### DISCUSSION THREAD

**Discussion by sschnug** [rules-proto-grpc/rules_proto_grpc#229] [Q&A] [answered]
Hi,

while i'm mostly using code-generation features (`python_grpc_compile` and co.) to prepare (dummy) packages (of generated code) for non-bazel consumers, i introduced some *canary-build* based on `cpp_grpc_library` for early detection of missing dependencies (leading to include-errors).

As this **target** has quite some effect on build-time, i'm **trying to turn it off by default and allow to enable explicitly** (for use in CI). 
Researching lead me to `target_compatible_with` approaches which seem to be fitting.

Do the rules of this project, e.g. `cpp_grpc_library`, support `target_compatible_with`?`
_I looks, like they don't_

The following should demonstrate the issue:

```starlark
load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
bool_flag(
    name = "flag",
    build_setting_default = False,
    visibility = ["//visibility:public"],
)

config_setting(
    name = "with_canary",
    flag_values = {
        ":flag": "true",
    },
)

# working 
# built with bazelisk build --//:flag //...
# not built with bazelisk build //...

cc_binary(
    name = 'test',
    srcs = ['test.cc'],
    target_compatible_with = select({
        ":with_canary": [],
        "//conditions:default": ["@platforms//:incompatible"],
    }),
)

# not working
# always built

cpp_grpc_library(
    name = "xxx-api-grpc-canary",
    protos = code_gen_deps_xxx_api + 
            code_gen_deps_yyy_api,
    target_compatible_with = select({
        ":with_canary": [],
        "//conditions:default": ["@platforms//:incompatible"],
    }),
)
```

**Comment by aaliddell** [comment, marked as answer]
I think this is related to #218, where some more attrs need propagating within the library macro. I'll be adding this for 4.3.0 in #227

**Reply by sschnug** [reply]
Looks good with 4.3.0.

Thank you!
